src/anndata/_io/specs/methods.py

```diff
@@ -301,13 +330,22 @@ def write_raw(f, k, raw, _writer, dataset_kwargs=MappingProxyType({})):
 
 @_REGISTRY.register_read(H5Group, IOSpec("dict", "0.1.0"))
 @_REGISTRY.register_read(ZarrGroup, IOSpec("dict", "0.1.0"))
-def read_mapping(elem, _reader):
+def read_mapping(
+    elem: GroupStorageType, *, _reader: Reader
+) -> dict[str, InMemoryArrayOrScalarType]:
```

Comment:
could also be recursive, right?

Maybe something like

```py
RWAble = InMemoryArrayOrScalarType | "RWAbleDict" | "RWAbleList"
RWAbleDict = dict[str, RWAble]
RWAbleList = list[RWAble]
```

Reply:
I actually don't know.  Can something like `obsm` be recursive?

Reply:
<img width="1512" alt="Screenshot 2024-07-11 at 11 33 58" src="https://github.com/scverse/anndata/assets/43999641/c6114a9f-4a29-41cd-9f16-b980e65e9e68">

Reply:
I guess this can be used for `uns` as well...